### DISCUSSION TITLE
Make close POIs visible

### DISCUSSION THREAD

**Discussion by j13m126** [organicmaps/organicmaps#1184] [Ideas]
When I'm adding new POIs (like restaurants and shops) to OpenStreetMap, I always check in Organic Maps if it already exists. It sometimes happens with several POIs in small places that not all are shown. This might happen because the house is too small and/or the POIs were set too close to each other. I also know that some commemorative plaques (and Stolpersteine) are placed as close as possible to each other and are therefore not all visible. Click [here](https://omaps.app/84GXDfcLbA/Währinger_Straße%2C_108) or [here](https://osmand.net/go?lat=48.22617&lon=16.34523&z=19), to see that not all POIs are visible. Same problem [here](https://omaps.app/84GXB3-gko/Gerta_Freimark), where only one out of four memorials is shown.

In OsmAnd you can click on details (of any POI) and see "Nearby POI". I honestly don't like this solution (because I think it should be more visible) but at least you can see all that is there. I would prefer a Popup where the user can select the POIs that are within a certain vicinity.

Could you please consider adding a feature that would make really close POIs visible?

Thank you and congratulations on this great APP!

**Comment by biodranik** [comment]
It is a great idea to make all nearby POIs visible. The hardest part is how to do it properly and conveniently for users.

OSMand approach is bad because you don't know that there are some hidden POIs when you are looking on the map.

We need to think about a better way to see POIs (or to hint that there are some other POIs in that place) directly from the map, without polluting it. One of the possible ideas is to group POIs, and hint with some different shape or color that several POIs are there.

Please share your ideas on how to make it in the best possible way.

**Comment by j13m126** [comment]
I agree. Their approach also takes too many clicks.

Following your idea of grouping POIs, I was thinking about 3 different symbols.
1. filled black circle with three white dots in the centre.
2. a circle, filled with four colours of POIS (brown, beige, violet and red)
3. a stack of circles

Clicking the grouping symbol could either zoom in until all POIs are separated, or show a list of POIs.

Another solution to indicate that not all POIs are visible would be a button saying "show all" which would only appear when zoomed in completely and there are hidden POIs. Clicking the button could lead to a list of POIs.

**Reply by patepelo** [reply]
Maximum zoom was improved for the next release, so you will be able to see the ones that are close to each other. 

Please tell us if is not enough so we can keep the discussion open and look for further solutios.
